src/ice/agent.rs

```diff
@@ -466,17 +467,35 @@ impl IceAgent {
 
     /// Adds a local candidate.
     ///
-    /// Returns `false` if the candidate was not added because it is redundant.
+    /// Returns `Result` if the candidate was added and `Err` in all other cases.
+    /// If the candidate was added, it should be signalled to the remote party.
+    ///
     /// Adding loopback addresses or multicast/broadcast addresses causes
     /// an error.
-    pub fn add_local_candidate(&mut self, mut c: Candidate) -> bool {
+    ///
+    /// The resulting error is boxed. To pattern match on it, dereference it first.
+    ///
+    /// ```
+    /// # use str0m::{Candidate, ice::IceAgent, error::IceError};
+    ///
+    /// let mut agent = IceAgent::new();
+    ///
+    /// match agent
+    ///     .add_local_candidate(Candidate::host("1.1.1.1:1000".parse().unwrap(), "udp").unwrap())
+    ///     .map_err(|e| *e) // Dereference boxed error to allow pattern matching.
+    /// {
+    ///     Ok(_) => {}
+    ///     Err(IceError::RedundantCandidate { .. }) => {}
+    ///     Err(IceError::NonHostCandidateInIceLite(..)) => {}
+    /// }
+    /// ```
+    pub fn add_local_candidate(&mut self, mut c: Candidate) -> Result<&Candidate, Box<IceError>> {
```

Comment:
Having to `Box` the error here is not ideal. Clippy was complaining that it is a very large error variant (168 bytes). Boxing it was the easiest to fix that as otherwise, I'd have to put an `#[allow]` on all call-sites that return `IceError` and `RtcError` (which embeds an `IceError`).

I added a doc-test to explain how to still pattern match on it. Open to other solutions!

Reply:
I'd hugely prefer `Option<&Candidate>`